packages/build/src/barque.ts

```diff
@@ -0,0 +1,137 @@
+import BuildVariant from './build-variant';
+import child_process from 'child_process';
+import Platform from './platform';
+import Config from './config';
+import util from 'util';
+import path from 'path';
+
+const execFile = util.promisify(child_process.execFile);
+/**
+ * Distro enum to be used when making a curator call.
+ */
+enum Distro {
+  Ubuntu = 'ubuntu1804',
+  Debian = 'debian10',
+  Redhat = 'rhel80'
+}
+
+/**
+ * Arch enum to be used when making a curator call.
+ *
+ * If we were to target a different arch for these distros, make sure
+ * config/repo-config.yml and packages/build/src/tarball.ts are changed accordingly.
+ */
+enum Arch {
+  Ubuntu = 'amd64',
+  Debian = 'amd64',
+  Redhat = 'x86_64',
+}
+
+export class Barque {
+  private config: Config;
+  private mongodbEdition: string;
+  private mongodbVersion: string;
+  private arch: string;
+
+  constructor(config: Config) {
+    this.config = config;
+    // hard code mongodb edition to 'org' for now
+    this.mongodbEdition = 'org';
+    // linux mongodb versions to release to. This should perhaps be an array of
+    // [4.3.0, 4.4.0], like mongo-tools
+    this.mongodbVersion = '4.4.0';
+  }
+
+  /**
+   * Upload current package to barque, MongoDB's PPA for linux distros.
+   *
+   * @param {string} tarballURL- The uploaded to Evergreen tarball URL.
+   * @param {Config} config - Config object.
+   *
+   * @returns {Promise} The promise.
+   */
+  async releaseToBarque(tarballURL: string): Promise<any> {
+    const repoConfig = path.join(this.config.rootDir, 'config', 'repo-config.yml');
+
+    if (this.config.platform === Platform.Linux) {
+      try {
+        await this.execCurator(tarballURL, repoConfig);
```

Comment:
try/catching this here so I can test this function easier. Still throw an error in the catch, so we can abort the release process if this is unsuccessful.